lib/actions.js

```diff
@@ -637,12 +637,10 @@ exports.upload = function(selector, path) {
 	var self = this;
 	if (fs.existsSync(path)) {
 		return new HorsemanPromise(function(resolve, reject) {
-			return self.page.evaluate(function(selector) {
-				self.page.uploadFile(selector, path, impatient(function() {
-					resolve();
-					debug(".upload() " + path + " into " + selector);
-				}, self.options.timeout));
-			});
+            return self.page.uploadFile(selector, path, function() {
+                resolve();
+                debug(".upload() " + path + " into " + selector);
+            });
 		});
```

Comment:
True, my change wouldn't actually stop the upload when the timeout was exceeded. It would just reject the Promise.

But it clearly looks like @johntitus intended for the upload to timeout after `self.options.timeout`. You probably shouldn't just ignore the timeout.